src/fu-engine.c

```diff
@@ -2937,6 +2960,45 @@ fu_engine_load_plugins (FuEngine *self, GError **error)
 		fu_plugin_list_add (self->plugin_list, plugin);
 	}
 
+	return TRUE;
+}
+
+static gboolean
+fu_engine_load_plugins (FuEngine *self, GError **error)
+{
+	g_autoptr(AsProfileTask) ptask = NULL;
+	g_autoptr(GPtrArray) paths = NULL;
+	GPtrArray *plugins;
+
+	/* profile */
+	ptask = as_profile_start_literal (self->profile, "FuEngine:load-plugins");
+	g_assert (ptask != NULL);
+
+	/* get a list of all plugin paths */
+	paths = fu_engine_get_plugin_paths (self);
+	if (paths->len == 0) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No plugin search paths found");
+		return FALSE;
+	}
+
+	/* look for all plugins */
+	for (guint i = 0; i < paths->len; i++) {
+		const gchar *path = g_ptr_array_index (paths, i);
+		if (!g_file_test (path, G_FILE_TEST_IS_DIR))
+			continue;
+		g_debug ("using plugin path of %s", path);
+		if (!fu_engine_load_plugins_for_path (self, path, error))
+			return FALSE;
+		/* we found some plugins */
+		plugins = fu_plugin_list_get_all (self->plugin_list);
+		if (plugins->len > 0)
+			break;
```

Comment:
Is this right? If we have a ./foo.so we don't load the other plugins from the other locations? Can't we just do this in preference, i.e. if `fu-plugin-thunderbolt.so` is available in `./` and `PLUGINDIR` just ignore the latter?

Reply:
I see you ended up doing this instead. Let me elaborate my thought process though. If this is coming as part of a snap/flatpak then it's entirely possible the plugins on system are much older than those in snap/flatpak. It's hard to know if they'll "really" be compatible so they should only really be used as a last resort if nothing was shipped in snap/flatpak or someone just takes the binary from within tree to run it.

Reply:
If they're running in flatpak, the host /usr is completely invisible; the PLUGINDIR will be something in `/app`...